modules/deployment-scripts/aks-run-command/main.bicep

```diff
@@ -31,35 +32,39 @@ param commands array
 @description('A delay before the script import operation starts. Primarily to allow Azure AAD Role Assignments to propagate')
 param initialScriptDelay string = '120s'
 
-@allowed([
-  'OnSuccess'
-  'OnExpiration'
-  'Always'
-])
+@allowed([ 'OnSuccess', 'OnExpiration', 'Always' ])
 @description('When the script resource is cleaned up')
 param cleanupPreference string = 'OnSuccess'
 
-resource aks 'Microsoft.ContainerService/managedClusters@2022-09-01' existing = {
+@description('Set to true when deploying template across tenants') 
```

Comment:
Can you expand on this scenario description further?

Reply:
When going cross-tenant, within a Marketplace application, without this the Marketplace's SP tries to run the custom script step and we can't RBAC failures.

We have tried without but have found that it is needed.